src/main/java/org/prebid/server/bidadjustments/BidAdjustmentsRetriever.java

```diff
@@ -0,0 +1,99 @@
+        this.mapper = Objects.requireNonNull(mapper).mapper();
+        this.jsonMerger = Objects.requireNonNull(jsonMerger);
+        this.samplingRate = samplingRate;
+    }
+
+    public BidAdjustments retrieve(AuctionContext auctionContext) {
+        final List<String> debugWarnings = auctionContext.getDebugWarnings();
+        final boolean debugEnabled = auctionContext.getDebugContext().isDebugEnabled();
+
+        final JsonNode requestBidAdjustmentsNode = Optional.ofNullable(auctionContext.getBidRequest())
+                .map(BidRequest::getExt)
+                .map(ExtRequest::getPrebid)
+                .map(ExtRequestPrebid::getBidadjustments)
+                .orElseGet(mapper::createObjectNode);
+
+        final JsonNode accountBidAdjustmentsNode = Optional.ofNullable(auctionContext.getAccount())
+                .map(Account::getAuction)
+                .map(AccountAuctionConfig::getBidAdjustments)
+                .orElseGet(mapper::createObjectNode);
+
+        final JsonNode mergedBidAdjustmentsNode = jsonMerger.merge(
+                requestBidAdjustmentsNode,
+                accountBidAdjustmentsNode);
+
+        try {
+            final ExtRequestBidAdjustments mergedBidAdjustments = mapper.convertValue(
+                    mergedBidAdjustmentsNode,
+                    ExtRequestBidAdjustments.class);
+
+            BidAdjustmentRulesValidator.validate(mergedBidAdjustments);
+            return BidAdjustments.of(mergedBidAdjustments);
+
+        } catch (IllegalArgumentException | ValidationException e) {
+            final String message = "bid adjustment from request was invalid: " + e.getMessage();
+            if (debugEnabled) {
+                debugWarnings.add(message);
+            }
+            conditionalLogger.error(message, samplingRate);
+            return retrieveAccountAdjustmentsOnly(accountBidAdjustmentsNode, debugEnabled, debugWarnings);
+        }
+    }
+
+    private BidAdjustments retrieveAccountAdjustmentsOnly(JsonNode accountBidAdjustmentsNode,
+                                                          boolean debugEnabled,
+                                                          List<String> debugWarnings) {
+
+        try {
+            final ExtRequestBidAdjustments accountBidAdjustments = mapper.convertValue(
+                    accountBidAdjustmentsNode,
+                    ExtRequestBidAdjustments.class);
+
+            BidAdjustmentRulesValidator.validate(accountBidAdjustments);
+            return BidAdjustments.of(accountBidAdjustments);
+        } catch (IllegalArgumentException | ValidationException e) {
+            final String message = "bid adjustment from account was invalid: " + e.getMessage();
+            if (debugEnabled) {
+                debugWarnings.add(message);
+            }
+            conditionalLogger.error(message, samplingRate);
+            return BidAdjustments.of(Collections.emptyMap());
+        }
```

Comment:
Duplicated code. Consider to  rewrite it like this:
```
final List<String> resolvedWarnings = debugEnabled ? debugWarnings : null; 
return convertAndValidate(mergedBidAdjustmentsNode, resolvedWarnings, "Error message")
                 .or(() -> convertAndValidate(accountBidAdjustmentsNode, resolvedWarnings, "Error message 2"))
                 .orElse(BidAdjustments.of(Collections.emptyMap()))
```
where `convertAndValidate ` will return `Optional`